src/codewhisperer/service/inlineCompletionService.ts

```diff
@@ -240,6 +242,29 @@ export class InlineCompletionService {
         return this.documentUri?.fsPath
     }
 
+    private sharedTryShowRecommendation = shared(this.tryShowRecommendation.bind(this))
+
+    private startShowRecommendationTimer() {
```

Comment:
This is a non-trivial pattern that is copy-pasted from other parts of the code. It might make sense to have a single shared implementation instead. (Could be done in a separate PR.)

Reply:
yes it make sense for this to be extract into a shared implementation. Will do that in another PR